yaksh/views.py

```diff
@@ -954,3 +956,75 @@ def design_questionpaper(request):
         context = {'form': form, 'questionpaper':True}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+@login_required
+def search_teacher(request, course_id):
+    """ search teachers for the course """
+
+    user = request.user
+    ci = RequestContext(request)
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+
+    context = {}
+    course = get_object_or_404(Course, creator=user, pk=course_id)
+    context['course'] = course
+
+    if request.method == 'POST':
+        u_name = request.POST.get('uname')
+        if len(u_name) == 0:
+            return my_render_to_response('yaksh/addteacher.html', context,
+                                        context_instance=ci)
+        else:
+            teachers = User.objects.filter(Q(username__icontains=u_name)|
```

Comment:
Everytime I search I will get the users who are already added to the course.
Can you filter the result before displaying